vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java

```diff
@@ -390,7 +392,8 @@ public void setCancelButton(Component component) {
      *            the element to display instead of default Cancel button
      */
     public void setCancelButton(Element element) {
-        addToSlot("cancel-button", element);
+        SlotUtils.clearSlot(this, "cancel-button");
+        SlotUtils.addToSlot(this, "cancel-button", element);
```

Comment:
nit: Would it make sense to capsulate both calls in a `SlotUtils.replaceSlot` or some other name? This would make sure that it's not forgotten by accident.

Reply:
Good point, I think we can add it 👍

Reply:
This PR seems to be in a good shape, so I'll create a separate PR for `replaceSlot` helper.

BTW we might want to also move the following logic for `Text` check to `SlotUtils`:

https://github.com/vaadin/flow-components/blob/35154825bca56b8b91dc0e92e7e3178ff0d0bfb7/vaadin-tabs-flow-parent/vaadin-tabs-flow/src/main/java/com/vaadin/flow/component/tabs/TabSheet.java#L297-L300

Reply:
Calling it "setSlot" would probably match the semantic :)

Reply:
Created a follow-up PR to implement this improvement: #4449